packages/react-native/ios/BdReactNative.mm

```diff
@@ -35,31 +47,13 @@ @implementation BdReactNative
   ];
 }
 
-RCT_EXPORT_METHOD(getDeviceID:(RCTPromiseResolveBlock)resolve
-                  rejecter:(RCTPromiseRejectBlock)reject)
-{
-  [CAPRNLogger getDeviceID:^(NSString *deviceID) {
-    if (deviceID == nil || [deviceID isEqual:[NSNull null]] || [deviceID isEqualToString:@""]) {
-      NSError *error = [NSError errorWithDomain:@"CAPRNLogger"
-                                           code:404
-                                       userInfo:@{NSLocalizedDescriptionKey: @"Device ID is undefined"}];
-      reject(@"device_id_undefined", @"Device ID is undefined", error);
-    } else {
-      resolve(deviceID);
-    }
-  } rejecter:^(NSString *code, NSString *message, NSError *error) {
-    reject(code, message, error);
-  }];
-}
-
-
 #else
 
 RCT_EXPORT_METHOD(init:(NSString*)apiKey
   sessionStrategy:(NSString*)sessionStrategy
   options:(JS::NativeBdReactNative::InitOptions& )options)
 {
-  BOOL enableNetworkInstrumentation = options != nil && options.enableNetworkInstrumentation().has_value() ? options.enableNetworkInstrumentation().value() : false;
```

Comment:
This null handling wasn't working right, so I moved the handling to the JS layer where we just ?? {}